### ISSUE TITLE
Training Loss become nan when do the shakespeare experiment under distributed environment

### ISSUE THREAD

**Issue by iuserea** [FedML-AI/FedML#60]
Code:
    just insert one line  of code in the file FedML/fedml_experiments/distributed/fedavg/mian_fedavg.py as shown in the fig1 below, others are just based on the latest **orgin/master.**
![image](https://user-images.githubusercontent.com/27194784/98211994-e3e8f400-1f7d-11eb-82c5-697043e0c42a.png)

Cmd:
sh run_fedavg_distributed_pytorch.sh 10 10 1 8 rnn hetero 100 10 10 0.8 shakespeare "./../../../data/shakespeare" 0

Result:
![image](https://user-images.githubusercontent.com/27194784/98211229-b3548a80-1f7c-11eb-9776-0f793f60d19d.png)

wandb:
![image](https://user-images.githubusercontent.com/27194784/98212854-2959f100-1f7f-11eb-97f9-5151128e1bcc.png)

Quesion:
    From wandb's screen shot,we can also notice that train/acc and test/acc are not so high.
    Is it normal?Does your guys have any advice to solve this?
    Thanks for your attention.

**Comment by chaoyanghe** [comment]
@iuserea It seems your non-IID version has lower accuracy. For this case, you need to turn your hyper-parameters. You can use grid search to find a better learning rate, batch size, local epoch, rounds.

**Comment by chaoyanghe** [comment]
@iuserea have you resolved your issue?

**Comment by iuserea** [comment]
@chaoyanghe I can't thank you enough! After trying grid search and random search,the acc became 40%.

**Comment by chaoyanghe** [comment]
@iuserea sounds good. Can you get results reported from this paper: https://arxiv.org/pdf/2003.00295.pdf ?

**Comment by iuserea** [comment]
@chaoyanghe it's possible to get results from this paper.I've read it before.But the acc there is about 57% which is more than 40% of mine.
